src/components/context/TodosContext.tsx

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { useEffect, useMemo, useState } from 'react';
+
+import * as todosServices from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { useError } from './ErrorContext';
+import { Status, TodoError } from '../../types/enums';
+
+interface TodosContextType {
+  todos: Todo[];
+  statusTodo: Status;
+  inputTodo: string;
+}
+
+const contextValue = {
+  todos: [],
+  statusTodo: Status.All,
+  inputTodo: '',
+};
+
+export const TodosContext = React.createContext<TodosContextType>(contextValue);
+
+export const TodosControlContext = React.createContext({
+  removeTodo: (_todoId: number) => {},
+  addTodo: (_todo: Todo) => {},
+  handleCheck: (_updatedTodo: Todo) => {},
+  handleClearAll: () => {},
+  toggleAll: () => {},
+  setStatusTodo: (_statusTodo: Status) => {},
+  setInputTodo: (_inputTodo: string) => {},
+});
+
+export function getMaxTodoId(todos: Todo[]) {
+  const ids = todos.map(todo => todo.id);
+
+  return Math.max(...ids, 0);
+}
+
+interface Props {
+  children: React.ReactNode;
+}
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [inputTodo, setInputTodo] = useState('');
+  const [statusTodo, setStatusTodo] = useState(Status.All);
+
+  const { setErrorMessage } = useError();
+  const [, setIsLoading] = useState(false);
```

Comment:
🤷‍♂️ 

For future tasks you can store an array of todo ids that currently loading and depending on this array show loaders for the corresponding todos